include/plugin/runtime-support.h

```diff
@@ -270,22 +274,54 @@ void confirm_protocol();
  */
 void reject_protocol(const std::string& reason);
 
+/**
+ * Opaque handle to a protocol analyzer.
+ */
+class ProtocolHandle {
+public:
+    ProtocolHandle() {}
+    ProtocolHandle(uint64_t id) : _id(id) {}
+
+    uint64_t id() const {
+        if ( ! _id )
+            throw ValueUnavailable("uninitialized protocol handle");
```

Comment:
This shouldn't really be happening because we're wrapping this into `optional`, right? Then it would at most be an internal error I'd say.

Reply:
Since `ProtocolHandle` is a rt type it needs to be default constructible. Such a default-constructed handle has no meaningful ID and we need to reject them.